documentation/library_guide/kernel_templates/esimd/radix_sort_by_key.rst

```diff
@@ -0,0 +1,360 @@
+Radix Sort By Key
+#################
+
+------------------------------------
+radix_sort_by_key Function Templates
+------------------------------------
+
+The ``radix_sort_by_key`` function sorts data using the radix sort algorithm.
+The sorting is stable, ensuring the preservation of the relative order of elements with equal keys.
+The functions implement a Onesweep* [#fnote1]_ algorithm variant. Both in-place and out-of-place
+overloads are provided. For out-of-place overloads, the input data order is preserved.
+
+A synopsis of the ``radix_sort_by_key`` function is provided below:
+
+.. code:: cpp
+
+   // defined in <oneapi/dpl/experimental/kernel_templates>
+
+   namespace oneapi::dpl::experimental::kt::gpu::esimd {
+
+   // Sort a sequence of keys and apply the same order to a sequence of values in-place
+   template <bool IsAscending = true, std::uint8_t RadixBits = 8,
+             typename KernelParam, typename Iterator1, typename Iterator2>
+   sycl::event
+   radix_sort_by_key (sycl::queue q, Iterator1 keys_first, Iterator1 keys_last,
+                      Iterator2 values_first, KernelParam param) // (1)
+
+   template <bool IsAscending = true, std::uint8_t RadixBits = 8,
+             typename KernelParam, typename KeysRng, typename ValuesRng>
+   sycl::event
+   radix_sort_by_key (sycl::queue q, KeysRng&& keys,
+                      ValuesRng&& values, KernelParam param) // (2)
+
+
+   // Sort a sequence of keys and values out-of-place
+   template <bool IsAscending = true, std::uint8_t RadixBits = 8,
+             typename KernelParam, typename KeysIterator1,
+             typename ValsIterator1, typename KeysIterator2,
+             typename ValsIterator2>
+   sycl::event
+   radix_sort_by_key (sycl::queue q, KeysIterator1 keys_first,
+                      KeysIterator1 keys_last, ValsIterator1 vals_first,
+                      KeysIterator2 keys_out_first, ValsIterator2 vals_out_first,
```

Comment:
Please change `vals` to `values` across the whole page.

Reply:
Done